### ISSUE TITLE
Implementation of an equality method `__eq__`.

### ISSUE THREAD

**Issue by eramdiaz** [lf1-io/padl#343]
## Description
At the moment there is not a customized `__eq__` method implemented on `padl`. It would be useful for the code and the tests, since it would allow us to compare two `Transform` `t` and `g` like `t == g`.

1. Design a similarity criterium: check what attributes of a `Transform` determine if they are equal or different.
2. Make sure that there are no conflicts between out `__eq__` method and the `torch` `__eq__` or other built-in `__eq__`.
3. Implement a `__hash__` method, required if you define `__eq__` method. Otherwise, we get a `TypeError` when calling `hash(<some_transform>)`

**Comment by wuhu** [comment]
On second thought, I think this is problematic:
Every callable can be a Transform, thus we can't really tell what equality should be every case and it would be impossible to implement this in a way that would be universally correct.
@eramdiaz I agree this would be useful for tests. Perhaps we can add an utility function with the tests to do this?

**Comment by eramdiaz** [comment]
yes, that is a good point @wuhu. So I we should limit ourselves to a utility function in tests, where all the transforms are predefined and we can make specific equality operations work, right?

**Comment by eramdiaz** [comment]
Should we then keep this open @wuhu ?

**Comment by wuhu** [comment]
Sorry, I missed your response, I think a utility function in tests would be useful.
